imagelab/ImgProvider.java

```diff
@@ -465,7 +470,9 @@ public void play() {
         }//for row
         int[] instruments = {Note.Vibes, Note.Pizzacatto, Note.MelodicTom};
         Music m = new Music(3, instruments);
-        m.playTune(tune);
+        m.playTune(tune, this);
+        });
+        playThread.start();
```

Comment:
Starts the thread with the play method in it given a playThread is not already running play.